org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/BaseLaunchRequestJob.java

```diff
@@ -137,6 +139,16 @@ private String toNonEmpty(List<String> stringValues, String defaultMessage) {
         return string != null ? string : defaultMessage;
     }
 
+    private BuildConfiguration buildConfigurationFor(RunConfiguration runConfig) {
```

Comment:
This is confusing, because we are using `BuildConfiguration` as a parameter holder here, which it is not intended for. Instead we should have:

```
BuildConfiguration {
  GradleArguments getGradleArguments();
}

RunConfiguration {
  GradleArguments getGradleArguments();
}

GradleArguments {
  void describeTo(Writer)
  void applyTo(GradleConnector)
  void applyTo(LongRunningOperation, BuildEnvironment)
}
```